### DISCUSSION TITLE
[ORM][OneToOne] Using relation with queries works slowly

### DISCUSSION THREAD

**Discussion by szal1k** [symfony/symfony#53955] [Q&A] [answered]
Hi,
I have problem with ORM\OneToOne relation. In my project exists two entities:

1) Document, with field:
```
    #[ORM\OneToOne(mappedBy: 'document')]
    private ?Bill $bill = null;
```

2) Bill, with field:
```
    #[ORM\OneToOne(targetEntity: Document::class, inversedBy: 'bill')]
    #[ORM\JoinColumn(name: 'document_id', referencedColumnName: 'id')]
    private ?Document $document = null;
```

When I build query with QueryBuilder like this:
```
$builder
->select(['d', 'b'])
->from(Document::class, 'd')
->leftJoin('d.bill', 'b');
```
and execute it, in Symfony profiler i have many queries and queries run very slow. For each row of document execute query to get bill. With realations ManyToOne/OneToMany i don't have this problem. It is a problem of my configuration or normal work of OneToOne relation and i can't change it?

**Comment by stof** [comment, marked as answer]
A bidirectional OneToOne is not able to perform lazy-loading on the inverse side (your `document` table does have a `bill_id` so the only way to initialize `$bill` is to do a query `SELECT * FROM bill WHERE document_id = :id`).
When you care about performance, keep your OneToOne relations unidirectional.

**Reply by mesolaries** [reply]
Why this is an issue in the OneToOne relationship? In the ManyToOne relationship the inverse side also does not have an id. The query should be executed only if I call `$document->getBill()`
